src/primitives/block.cpp

```diff
@@ -11,55 +11,58 @@
 #include <crypto/common.h>
 
 // yespower
-#include <stdlib.h>
 #include <crypto/yespower-1.0.1/yespower.h>
 #include <streams.h>
 #include <version.h>
 
-// yespower cache
+// yespower exit()
+#include <stdlib.h>
+
+// yespower PoW cache
 #include <sync.h>
 
 uint256 CBlockHeaderUncached::GetHash() const
 {
     return SerializeHash(*this);
 }
 
-// yespower
+// yespowerUncached
 uint256 CBlockHeaderUncached::GetPoWHash() const
 {
-    uint256 thash;
-    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
-    ss << *this;
-    yespower_params_t yespower_1_0_sugarchain = {
+    static const yespower_params_t yespower_1_0_sugarchain = {
         .version = YESPOWER_1_0,
         .N = 2048,
         .r = 32,
         .pers = (const uint8_t *)"Satoshi Nakamoto 31/Oct/2008 Proof-of-work is essentially one-CPU-one-vote",
         .perslen = 74
     };
-    if (yespower_tls( (unsigned char *)&ss[0], ss.size(), &yespower_1_0_sugarchain, (yespower_binary_t *)&thash) ) {
-        abort();
+    uint256 hash;
+    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+    ss << *this;
+    if (yespower_tls((const uint8_t *)&ss[0], ss.size(), &yespower_1_0_sugarchain, (yespower_binary_t *)&hash)) {
```

Comment:
`const uint8_t` instead of `unsigned char` are you sure?

Reply:
need to check compiler log
https://stackoverflow.com/questions/7009231/cast-unsigned-char-uint8-t-to-const-char

Reply:
no compiler warning
https://github.com/sugarchain-project/sugarchain/issues/63